### ISSUE TITLE
Add more substantial caching to the recursor

### ISSUE THREAD

**Issue by bluejekyll** [hickory-dns/hickory-dns#1720]
**Is your feature request related to a problem? Please describe.**

Once  #1710 is merged, we should add a new caching authority that sits in front of the recursor requests and make sure it is suitable for large numbers of requests. Consider making the max and min TTLs be configurable by record type as well.

**Describe the solution you'd like**

The recursive resolver has an LRU integrated to day, but it is not currently configurable. The max and min TTLs for the LRU should be exposed to the config file in the server for the recursor (these should be added to the forwarder/resolver config as well to be consistent).

**Comment by divergentdave** [comment]
Is #2524 sufficient to close out this issue?

I think the recursor's internal LRU cache is preferable to the idea of a separate caching authority from the first paragraph. If the cache and the recursor were separated by a layer of encapsulation, that would preclude caching internally-generated queries for nameserver referrals, etc.

The forwarder/resolver config currently includes minimum and maximum TTL values. I did not add the ability to change these for particular record types to the resolver config, as I did to the recursor config. If there's interest in doing so, that should be straightforward, though we'll have to decide what the configuration should look like, and how to handle backwards compatibility of `positive_min_ttl`, etc. at the top-level of `ResolverOpts`.
